Swerve2022/src/main/java/frc/robot/subsystems/SwerveModule.java

```diff
@@ -42,36 +42,13 @@
 
   /**
    * Constructs a SwerveModule with a drive motor, turning motor, drive encoder and turning encoder.
-   *
-   * @param driveMotorChannel
-   *          PWM output for the drive motor.
-   * @param turningMotorChannel
-   *          PWM output for the turning motor.
-   * @param driveEncoderChannelA
-   *          DIO input for the drive encoder channel A
-   * @param driveEncoderChannelB
-   *          DIO input for the drive encoder channel B
-   * @param turningEncoderChannelA
-   *          DIO input for the turning encoder channel A
-   * @param turningEncoderChannelB
-   *          DIO input for the turning encoder channel B
    */
-  public SwerveModule(int driveMotorCANID, int turningMotorCANID, int driveEncoderChannelA, int driveEncoderChannelB,
-      int turningEncoderChannelA, int turningEncoderChannelB)
+  public SwerveModule(int driveMotorCANID, int turningMotorCANID, int addressCANID)
   {
     m_driveMotor = new WPI_TalonFX(driveMotorCANID);
     m_turningMotor = new WPI_TalonFX(turningMotorCANID);
 
-    m_driveEncoder = new Encoder(driveEncoderChannelA, driveEncoderChannelB);
-    m_turningEncoder = new Encoder(turningEncoderChannelA, turningEncoderChannelB);
-
-    // Set the distance per pulse for the drive encoder. We can simply use the distance
-    // traveled for one rotation of the wheel divided by the encoder resolution.
-    m_driveEncoder.setDistancePerPulse(2 * Math.PI * kWheelRadius / kEncoderResolution);
-
-    // Set the distance (in this case, angle) per pulse for the turning encoder. This is
-    // the the angle through an entire rotation (2 * pi) divided by the encoder resolution.
-    m_turningEncoder.setDistancePerPulse(2 * Math.PI / kEncoderResolution);
+    m_turningCANCoder = new CANCoder(addressCANID);
```

Comment:
When I looked at the CANCoder getAbsolutePosition method, it reports 0..360 degrees. But I think the Encoder class we replaced works +/- from 0. So the CANCoder needs to be configured o work -180 to 180. Look in the CTRE API docs to see how to do that no the CANcoder.